tests/unit/-private/properties/legacy-collection-test.js

```diff
@@ -0,0 +1,584 @@
+    let page = create({
+      foo: collection({
+        scope: 'div',
+        itemScope: 'span',
+
+        item: {
+          text: text()
+        }
+      })
+    });
+
+    this.adapter.createTemplate(this, page, `
+      <div>
+          <span>Lorem</span>
+          <span>Ipsum</span>
+      </div>
+    `);
+
+    let textContents = [];
+
+    for (let i = 0; i < page.foo().count; i++) {
+      let item = page.foo(i);
+      textContents.push(item.text);
+    }
+
+    assert.deepEqual(textContents, ['Lorem', 'Ipsum']);
+  });
+
+  test('iterates over scoped items with a for of loop', function(assert) {
+    let page = create({
+      foo: collection({
+        scope: 'div',
+        itemScope: 'span',
+
+        item: {
+          text: text()
+        }
+      })
+    });
+
+    this.adapter.createTemplate(this, page, `
+      <div>
+          <span>Lorem</span>
+          <span>Ipsum</span>
+      </div>
+    `);
+
+    let textContents = [];
+
+    withIteratorSymbolDefined(() => {
+      for (let item of page.foo()) {
+        textContents.push(item.text);
+      }
+    });
+
+    assert.deepEqual(textContents, ['Lorem', 'Ipsum']);
+  });
+
+  test('iterates over scoped items with a forEach loop', function(assert) {
+    let page = create({
+      foo: collection({
+        scope: 'div',
+        itemScope: 'span',
+
+        item: {
+          text: text()
+        }
+      })
+    });
+
+    this.adapter.createTemplate(this, page, `
+      <div>
+          <span>Lorem</span>
+          <span>Ipsum</span>
+      </div>
+    `);
+
+    let textContents = [];
+
+    page.foo().toArray().forEach(function(item) {
+      textContents.push(item.text);
+    });
+
+    assert.deepEqual(textContents, ['Lorem', 'Ipsum']);
+  });
+
+  test('does not mutate definition object', function(assert) {
+    let prop = text('.baz');
+    let expected = {
+      scope: '.a-scope',
+      itemScope: '.another-scope',
+      item: {
+        baz: prop
+      },
+
+      bar: prop
+    };
+    let actual = {
+      scope: '.a-scope',
+      itemScope: '.another-scope',
+      item: {
+        baz: prop
+      },
+
+      bar: prop
+    };
+
+    let page = create({
+      foo: collection(actual)
+    });
+
+    this.adapter.createTemplate(this, page);
+
+    assert.deepEqual(actual, expected);
+  });
+
+  test('looks for elements within test container specified', function(assert) {
+    assert.expect(2);
+
+    let expectedContext = '#alternate-ember-testing';
+    let page;
+
+    page = create({
+      foo: collection({
+        testContainer: expectedContext,
+        itemScope: 'span'
+      })
+    });
+
+    this.adapter.createTemplate(
+      this,
+      page,
+      '<span>Lorem</span><span>ipsum</span>',
+      { useAlternateContainer: true }
+    );
+
+    assert.equal(page.foo().count, 2);
+    assert.equal(page.foo(0).text, 'Lorem');
+  });
+
+  test('can provide custom array methods', function(assert) {
```

Comment:
I'm afraid even to think why this functionality is needed 😨

Reply:
Yeah, that's a great point...maybe we should not allow this.

Reply:
This is for the legacy collection only, so it is providing existing behavior. The reason I added the test was to get code coverage passing. 

Part of the failure was because of [this if statement here](https://github.com/pzuraq/ember-cli-page-object/blob/1402a5e69adc37d7c92b7b25ff360a2a6cbd517f/addon/-private/properties/collection/legacy.js#L30-L32). Because it is dependent on the window/browser, we cannot test the `else` case (that `Symbol` is undefined). Unfortunately, we had to add another similar `if` statement in the new implementation, which was also untestable, lowering code coverage overall. The only way I found around this was to add tests for previous behavior that was not tested, but did exist.

Reply:
Ah, right. The code coverage issue. So when we pull out the old collection & its tests, we'll get another coverage failure?

Reply:
I believe overall code coverage will actually go up at that point, because we'll be removing all the cases _and_ one of the `Symbol` if statements

Reply:
Sad but true.. I've never liked this partially supported `Symbol.Iterator` feature 😆